shared/modules/persistence_module.go

```diff
@@ -33,6 +38,22 @@ type PersistenceModule interface {
 	HandleDebugMessage(*messaging.DebugMessage) error
 }
 
+// TreeStore is fulfilled by the treeStore to create an
+// atomic tree component for use by the peristence context.
+type TreeStore interface {
+	// Update returns the new state hash for a given height.
+	// * Height is passed through to the Update function
+	// and used by the queries against the TxIndexer and the
+	// It updates to the future but not to the past.
+	// * Passing a higher height will cause a change
+	// but repeatedly calling the same or a lower height will
+	// not incur a change.
+	Update(pgtx pgx.Tx, txi indexer.TxIndexer, height uint64) (string, error)
```

Comment:
I have a lot of context (more than most) on the work you're doing here and it was hard for my to decipher the interface and implementation. What do you think of this updated interface (note that the txi is part of the state)?

```go
// Update uses the ephemeral state in the postgres transaction pgtx at the given height and automatically
// updates all the trees backing the persistence module. 
// Updates to current or prior heights are idempotent and return the same hash every time.
// Updates to future heights returns an error.
`Update(pgtx pgx.Tx, height uint64) (string, error)`
```

Reply:
@dylanlott Bump on this?

Reply:
I was going to address this in #808 because it seemed better suited to handle it with the other module changes. Is that reasonable? I can handle it here if preferred.